app/Models/Role.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Role extends Model
+{
+    protected $fillable = [
+        'name', 
+        'slug', 
+        'permissions',
+    ];
+
+    protected $casts = [
+        'permissions' => 'array',
+    ];
+
+    public function users()
+    {
+    	return $this->belongsToMany('App\Models\User', 'role_user');
+    }
+
+    public function hasAccess(array $permissions):bool
+    {
+        foreach($permissions as $permission)
+        {
+        	if($this->hasPermission($permission))
```

Comment:
`Cần phải có một dấu cách phía sau những từ khoá Control structure (như if, else, for ...)`

https://viblo.asia/p/cac-convention-thuong-gap-khi-moi-bat-dau-training-php-tai-framgia-bJzKmgq6l9N#_24-khoang-trang-whitespace-9